src/App.jsx

```diff
@@ -1,45 +1,96 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import { Routes, Route, Navigate, useNavigate, useLocation } from 'react-router-dom';
 import Header from './components/Header/Header';
 import Courses from './components/Courses/Courses';
 import CreateCourse from './components/CreateCourse/CreateCourse';
+import Registration from './components/Registration/Registration';
+import Login from './components/Login/Login';
 import './App.css';
-import { mockedCoursesList, mockedAuthorsList } from './constants';
+import { api_endpoints, storageKeys, paths } from './constants';
 
 const App = () => {
-  const [courses, setCourses] = useState(mockedCoursesList);
-  const [authors, setAuthors] = useState(mockedAuthorsList);
-  const [isCreatingCourse, setIsCreatingCourse] = useState(false);
+  const [courses, setCourses] = useState([]);
+  const [authors, setAuthors] = useState([]);
+  const [userName, setUserName] = useState(localStorage.getItem(storageKeys.userName) || '');
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  useEffect(() => {
+    const token = localStorage.getItem(storageKeys.userToken);
+    const isAuthPage = location.pathname === paths.login || location.pathname === paths.registration;
+
+    if (token && isAuthPage) {
+      navigate(paths.courses);
+    }
+  }, [navigate, location.pathname]);
+
+  useEffect(() => {
+    // Fetch courses
+    const fetchCourses = async () => {
```

Comment:
Minor, but I think the comments might not be necessary here since you named the function nicely 😅.
Same for fetch authors.